frontend/src/concepts/topology/useTopologyController.ts

```diff
@@ -16,7 +17,9 @@ const useTopologyController = (graphId: string): Visualization | null => {
   React.useEffect(() => {
     const visualizationController = new Visualization();
     visualizationController.setFitToScreenOnLayout(true);
+    visualizationController.registerElementFactory(pipelineElementFactory);
     visualizationController.registerComponentFactory(pipelineComponentFactory);
+    visualizationController.registerComponentFactory(pipelineGroupsComponentFactory);
```

Comment:
If you are including `pipelineComponentFactory` then `pipelineGroupsComponentFactory` does not need to duplicate the types in its switch statement. PF will check all factories in reverse order added, so only overrides need to be in the later factories.

If you add the PF pipelines factory (first), you could also remove the `DEFAULT_SPACER_NODE_TYPE` and `DEFAULT_EDGE_TYPE` cases.

Reply:
Great point, thank you @jeff-phillips-18 . I've addressed this comment and your suggestions above in https://github.com/opendatahub-io/odh-dashboard/pull/2725